expression/column.go

```diff
@@ -223,6 +223,20 @@ func (col *CorrelatedColumn) RemapColumn(m map[int64]*Column) (Expression, error
 	}, nil
 }
 
+// HashCode implements Expression interface.
+func (col *CorrelatedColumn) HashCode(sc *stmtctx.StatementContext) []byte {
+	// Doesn't check len(hashcode) because col.Data can be changed anytime,
+	// So always use newest Datum to calc hash code.
+	col.hashcode = make([]byte, 0, 9)
```

Comment:
Each call of `(*CorrelatedColumn) HashCode) will create a new slice, because we don't know if the datum `col.Data` pointed changed or not.

Reply:
Refined